scripts/get-venue-users-reports.ts

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env node -r esm -r ts-node/register
+import { initFirebaseAdminApp, makeScriptUsage } from "./lib/helpers";
+import admin from "firebase-admin";
+import * as path from "path";
+import * as fs from "fs";
+import { FormatterOptionsArgs, Row, writeToStream } from "@fast-csv/format";
+
+type CsvFileOpts = {
+  headers: string[];
+  path: string;
+};
+
+class CsvFile {
+  static write(
+    stream: NodeJS.WritableStream,
+    rows: Row[],
+    options: FormatterOptionsArgs<Row, Row>
+  ): Promise<void> {
+    return new Promise((res, rej) => {
+      writeToStream(stream, rows, options)
+        .on("error", (err: Error) => rej(err))
+        .on("finish", () => res());
+    });
+  }
+
+  private readonly headers: string[];
+
+  private readonly path: string;
+
+  private readonly writeOpts: FormatterOptionsArgs<Row, Row>;
+
+  constructor(opts: CsvFileOpts) {
+    this.headers = opts.headers;
+    this.path = opts.path;
+    this.writeOpts = { headers: this.headers, quote: true };
+  }
+
+  create(rows: Row[]): Promise<void> {
+    return CsvFile.write(fs.createWriteStream(this.path), rows, {
+      ...this.writeOpts,
+    });
+  }
+}
+
+const usage = makeScriptUsage({
+  description: "Bulk upload events from a spreadsheet ",
+  usageParams: "PROJECT_ID CREDENTIAL_PATH",
+  exampleParams: "co-reality-staging add-event-by-csv.example.csv key.json",
+});
+
+const [projectId, credentialPath] = process.argv.slice(2);
+
+if (!credentialPath) {
+  usage();
+}
+
+initFirebaseAdminApp(projectId, {
+  credentialPath: credentialPath
+    ? path.resolve(__dirname, credentialPath)
+    : undefined,
+});
+
+const runner = async () => {
+  const allUsers: admin.auth.UserRecord[] = [];
+  let nextPageToken: string | undefined;
+  const { users, pageToken } = await admin.auth().listUsers(10);
+
+  allUsers.push(...users);
+  nextPageToken = pageToken;
+
+  while (nextPageToken) {
+    const { users, pageToken } = await admin
+      .auth()
+      .listUsers(1000, nextPageToken);
+    allUsers.push(...users);
+    nextPageToken = pageToken;
+  }
+
+  let headers = [
+    "Email",
+    "Party Name",
+    "Full name",
+    "Company title",
+    "Company department",
+    "Ask me about:",
+    "I'm loving this song - book - show: ",
+    "I've been at GitHub for:",
+    "The best part of my role at GitHub is:",
+    "When not working I'm often:",
+    "Last Sign In",
+  ];
+
+  const csvFile = new CsvFile({
+    path: path.resolve(
+      __dirname,
+      "users-report-" + new Date().getTime() + ".csv"
+    ),
+    headers: headers,
+  });
```

Comment:
Why do we need a whole class object here? We can just resolve the path to a `const`.